### PR TITLE
Create sample projects for AdaptiveLoadBalancingGroup

### LINKED ISSUE #4679: Cluster.Metrics Adaptive Load Balancer not collecting metrics
When using the Adaptive Load Balancer with a Cluster Group Router the metrics are not collected by the AdaptiveLoadBalancingMetricsListener unless a new router management message that is not handled by the ClusterRouterActor and the RouterActor is sent to the router,

`
       // RouterActor
       
        protected override void OnReceive(object message)
        {
            switch (message)
            {
                case GetRoutees getRoutees:
                    Sender.Tell(new Routees(Cell.Router.Routees));
                    break;
                case AddRoutee addRoutee:
                    Cell.AddRoutee(addRoutee.Routee);
                    break;
                case RemoveRoutee removeRoutee:
                    Cell.RemoveRoutee(removeRoutee.Routee, stopChild: true);
                    StopIfAllRouteesRemoved();
                    break;
                case Terminated terminated:
                    Cell.RemoveRoutee(new ActorRefRoutee(terminated.ActorRef), stopChild: false);
                    StopIfAllRouteesRemoved();
                    break;
                default:
                    // Any other unhandled management message wil be forwarded to the controller and start collecting metrics
                    RoutingLogicController?.Forward(message);
                    break;
            }
        }
`

- Akka 1.14.12
- dotnet core 3.1

### PR DESCRIPTION
**Pull request by Arkatufus**
Closes #4679

### CODE REVIEW

File: src/core/Akka/Configuration/Hocon/HoconValue.cs [hunk 1]
```diff
@@ -73,6 +73,11 @@ public bool IsEmpty
         /// </summary>
         internal bool AdoptedFromFallback { get; private set; }
 
+        public Config ToConfig()
+        {
+            return new Config(new HoconRoot(this, Enumerable.Empty<HoconSubstitution>()));
```
Comment (Aaronontheweb) [inline src/core/Akka/Configuration/Hocon/HoconValue.cs:78]: Is the idea here to perform a deep, immutable clone of the entire HOCON graph?

File: src/core/Akka/Configuration/Pigeon.conf [hunk 2]
```diff
@@ -142,6 +142,11 @@ akka {
           consistent-hashing-group = "Akka.Routing.ConsistentHashingGroup"
           tail-chopping-pool = "Akka.Routing.TailChoppingPool"
           tail-chopping-group = "Akka.Routing.TailChoppingGroup"
+
+          # Provide routing of messages based on cluster metrics.
+          # Akka.Cluster.Metrics needs to be installed to use these extensions
+          cluster-metrics-adaptive-pool = "Akka.Cluster.Metrics.AdaptiveLoadBalancingPool, Akka.Cluster.Metrics"
```
Comment (Aaronontheweb) [inline src/core/Akka/Configuration/Pigeon.conf:148]: Yep, this looks right based on what we discussed in chat

File: src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Application.conf [hunk 3]
```diff
@@ -0,0 +1,53 @@
+﻿akka 
+{
+    # Enable metrics extension in akka-cluster-metrics.
+    extensions = ["Akka.Cluster.Metrics.ClusterMetricsExtensionProvider, Akka.Cluster.Metrics"]
+
+    actor {
+        provider = cluster
+
+        deployment {
+            /factorialBackendRouter {
```
Comment (Aaronontheweb) [inline src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Application.conf:10]: LGTM

File: src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Program.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                {
+                    "/user/factorialBackend-1",
+                    "/user/factorialBackend-2",
+                    "/user/factorialBackend-3",
+                    "/user/factorialBackend-4",
+                    "/user/factorialBackend-5",
+                    "/user/factorialBackend-6"
+                };
+
+                system.ActorOf(
```
Comment (Aaronontheweb) [inline src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Program.cs:35]: for the sample here, I would do two routers:

1. Created entirely from HOCON;
2. One created entirely from the fluent interface on `Props`.

So you can demonstrate how both work - and I would decorate them with comments.

File: src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Samples.Cluster.AdaptiveGroup.csproj [hunk 5]
```diff
@@ -0,0 +1,20 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (Aaronontheweb) [inline src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Samples.Cluster.AdaptiveGroup.csproj:5]: Should consume `common.props` here and specify the target framework using that file - like how we do here:

https://github.com/akkadotnet/akka.net/blob/b7630e5bf6e81631ace79bf3db0de38818a09f02/src/examples/Cluster/Samples.Cluster.Simple/Samples.Cluster.Simple.csproj#L3-L7

Comment (Aaronontheweb) [reply src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Samples.Cluster.AdaptiveGroup.csproj:5]: Also - need to add `<IsPackable>false</IsPackable>` here so we don't accidentally create a NuGet package

File: src/examples/Cluster/Metrics/Samples.Cluster.Metrics.Common/MetricListener.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        public MetricListener()
+        {
+            _extension = ClusterMetrics.Get(Context.System);
+
+            var log = Context.GetLogger();
+            var cluster = Akka.Cluster.Cluster.Get(Context.System);
+
+            Receive<ClusterMetricsChanged>(clusterMetrics =>
+            {
+                clusterMetrics.NodeMetrics
```
Comment (Aaronontheweb) [inline src/examples/Cluster/Metrics/Samples.Cluster.Metrics.Common/MetricListener.cs:28]: This is cool

File: src/examples/Cluster/Metrics/Samples.Cluster.Metrics.Common/Samples.Cluster.Metrics.Common.csproj [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (Aaronontheweb) [inline src/examples/Cluster/Metrics/Samples.Cluster.Metrics.Common/Samples.Cluster.Metrics.Common.csproj:4]: Same feedback as above about `common.props`

File: src/examples/Cluster/Metrics/Samples.Cluster.Metrics/Samples.Cluster.Metrics.csproj [hunk 8]
```diff
@@ -0,0 +1,20 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (Aaronontheweb) [inline src/examples/Cluster/Metrics/Samples.Cluster.Metrics/Samples.Cluster.Metrics.csproj:5]: Same feedback as above about `common.props`

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Some minor changes that need to be made as well as a clarification on the HOCON change

File: src/core/Akka/Configuration/Hocon/HoconValue.cs (hunk 1 above)
Comment (Arkatufus) [reply src/core/Akka/Configuration/Hocon/HoconValue.cs:78]: No, its just a convenience method to convert a `HoconValue` into a `Config`

File: src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Program.cs (hunk 4 above)
Comment (Arkatufus) [reply src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Program.cs:35]: I did it on this line: https://github.com/akkadotnet/akka.net/pull/4691/files/e2847bd2e4fa084e3acc06fa7a97f7e895524421#diff-d1fee1dcd9811e796dd22889fe7ce472e977f115f35b914ba002783920119dedR45

Comment (Aaronontheweb) [reply src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Program.cs:35]: can you add some comments on it?

Comment (Aaronontheweb) [reply src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Program.cs:35]: Thinking about it from the point of view of an end-user who isn't sure what the syntax means right off the bat

Comment (Arkatufus) [reply src/examples/Cluster/Metrics/Samples.Cluster.AdaptiveGroup/Program.cs:35]: Will do

**Review by Aaronontheweb — APPROVED** [review]
LGTM

File: src/examples/Cluster/Metrics/Samples.Cluster.Metrics.Common/Samples.Cluster.Metrics.Common.csproj [hunk 9]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="..\..\..\..\common.props" />
+
+  <PropertyGroup>
+    <TargetFramework>$(NetStandardLibVersion)</TargetFramework>
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (Aaronontheweb) [inline src/examples/Cluster/Metrics/Samples.Cluster.Metrics.Common/Samples.Cluster.Metrics.Common.csproj:10]: Have a duplicate `TargetFramework` attribute here

**Pull request merged**